java/com/jfreechartextensions/axistransformations/ExtendedNumberAxis.java

```diff
@@ -369,41 +377,41 @@ protected void autoAdjustRange() {
                 double upper = transform.transform(r.getUpperBound());
                 double lower = transform.transform(r.getLowerBound());
 
-                if(this.getRangeType() == RangeType.POSITIVE) {
+                if (this.getRangeType() == RangeType.POSITIVE) {
                     lower = Math.max(0.0D, lower);
                     upper = Math.max(0.0D, upper);
-                } else if(this.getRangeType() == RangeType.NEGATIVE) {
+                } else if (this.getRangeType() == RangeType.NEGATIVE) {
                     lower = Math.min(0.0D, lower);
                     upper = Math.min(0.0D, upper);
                 }
 
-                if(this.getAutoRangeIncludesZero()) {
+                if (this.getAutoRangeIncludesZero()) {
                     lower = Math.min(lower, 0.0D);
                     upper = Math.max(upper, 0.0D);
                 }
 
                 final double range = upper - lower;
                 final double fixedAutoRange = this.getFixedAutoRange();
-                if(fixedAutoRange > 0.0D) {
+                if (fixedAutoRange > 0.0D) {
                     lower = upper - fixedAutoRange;
                 } else {
                     final double minRange = this.getAutoRangeMinimumSize();
-                    if(range < minRange) {
+                    if (Math.abs(range) < minRange) {
                         final double expand = (minRange - range) / 2.0D;
```

Comment:
I was wrong earlier. This line doesn't assume range to be positive, however, this line wants it to be positive to compare with `minRange`.

`range` can be negative/positive according to transformation applied and we want it to be as it is since we want the correct value to be added/subtracted while computing the margins on lines 433 and 434